scripts/gitOpsApi.js

```diff
@@ -0,0 +1,429 @@
+		let parentCommit;
+		const repoOwner = options.repoOwner;
+		const repoName = options.repoName;
+		const branchName = options.branchName;
+		const files = options.files;
+		const commitMsg = options.commitMsg;
+
+		/**
+		 * Returns a single tree using the SHA1 value for that tree.
+		 * @param {String} repoOwner - The repo owner.
+		 * @param {String} repoName - The repo name.
+		 * @param {String} branchName - The name of the remote branch to get the tree from.
+		 * @return {Promise} The resolve or reject of the promise indicates whether the operation completes successfully.
+		 */
+		const getTree = async function (repoOwner, repoName, branchName) {
+			return module.exports.getBranchRef(octokit, {
+				repoOwner: repoOwner,
+				repoName: repoName,
+				branchName: branchName
+			}).then((commit) => {
+				parentCommit = commit;
+				return octokit.request("GET /repos/{owner}/{repo}/git/trees/{tree_sha}", {
+					owner: repoOwner,
+					repo: repoName,
+					tree_sha: commit.object.sha
+				});
+			});
+		};
+
+		return new Promise((resolve, reject) => {
+			return Promise.all(files.map((file) => {
+				return octokit.request("POST /repos/{owner}/{repo}/git/blobs", {
+					owner: repoOwner,
+					repo: repoName,
+					content: file.content
+				});
+			})).then((blobs) => {
+				return getTree(repoOwner, repoName, branchName).then(function (tree) {
+					return octokit.request("POST /repos/{owner}/{repo}/git/trees", {
+						owner: repoOwner,
+						repo: repoName,
+						tree: files.map(function (file, index) {
+							return {
+								path: file.path,
+								mode: "100644",
+								type: "blob",
+								sha: blobs[index].data.sha
+							};
+						}),
+						base_tree: tree.data.sha
+					});
+				});
+			}).then((tree) => {
+				return octokit.request("POST /repos/{owner}/{repo}/git/commits", {
+					owner: repoOwner,
+					repo: repoName,
+					message: commitMsg,
+					tree: tree.data.sha,
+					parents: [parentCommit.object.sha]
+				});
+			}).then((commit) => {
+				return octokit.request("PATCH /repos/{owner}/{repo}/git/refs/{ref}", {
+					owner: repoOwner,
+					repo: repoName,
+					ref: "heads/" + branchName,
+					sha: commit.data.sha
+				}).then(() => {
+					resolve("Committed successfully.");
+				}).catch((e) => {
+					reject({
+						isError: true,
+						message: "Error at commitMultipleFiles(): " + e.message
+					});
+				});
+			}).catch((e) => {
+				reject({
+					isError: true,
+					message: "Error at commitMultipleFiles(): " + e.message
+				});
+			});
+		});
+	},
+
+	/**
+	 * An object that contains required information for submitting multiple files in one commit.
+	 * @typedef {Object} IssuePullRequestOptions
+	 * @param {String} repoOwner - The repo owner.
+	 * @param {String} repoName - The repo name.
+	 * @param {String} issuerGithubId - The Github id of the issuer.
+	 * @param {String} branchName - The name of the remote branch to operate.
+	 * @param {String} prTitle - The pull request title.
```

Comment:
It's perhaps a bit more typing, but I'd prefer fuller words in these option names since they will probably end up appearing exposed in lots of places to people who will have to recognise them easily, e.g. prTitle -> pullRequestTitle and commitMsg -> commitMessage.